sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/models/AppendBlobsAppendBlockHeaders.java

```diff
@@ -93,6 +95,40 @@ public final class AppendBlobsAppendBlockHeaders {
     @JsonProperty(value = "x-ms-encryption-scope")
     private String xMsEncryptionScope;
 
+    // HttpHeaders containing the raw property values.
+    /**
+     * Creates an instance of AppendBlobsAppendBlockHeaders class.
+     *
+     * @param rawHeaders The raw HttpHeaders that will be used to create the property values.
+     */
+    public AppendBlobsAppendBlockHeaders(HttpHeaders rawHeaders) {
+        this.xMsVersion = rawHeaders.getValue("x-ms-version");
+        if (rawHeaders.getValue("x-ms-content-crc64") != null) {
+            this.xMsContentCrc64 = Base64.getDecoder().decode(rawHeaders.getValue("x-ms-content-crc64"));
+        }
+        if (rawHeaders.getValue("x-ms-blob-committed-block-count") != null) {
+            this.xMsBlobCommittedBlockCount = Integer.parseInt(rawHeaders.getValue("x-ms-blob-committed-block-count"));
+        }
+        if (rawHeaders.getValue("Last-Modified") != null) {
+            this.lastModified = new DateTimeRfc1123(rawHeaders.getValue("Last-Modified"));
+        }
+        this.xMsEncryptionKeySha256 = rawHeaders.getValue("x-ms-encryption-key-sha256");
+        if (rawHeaders.getValue("x-ms-request-server-encrypted") != null) {
+            this.xMsRequestServerEncrypted = Boolean.parseBoolean(rawHeaders.getValue("x-ms-request-server-encrypted"));
+        }
+        if (rawHeaders.getValue("Date") != null) {
+            this.dateProperty = new DateTimeRfc1123(rawHeaders.getValue("Date"));
+        }
+        if (rawHeaders.getValue("Content-MD5") != null) {
+            this.contentMD5 = Base64.getDecoder().decode(rawHeaders.getValue("Content-MD5"));
+        }
+        this.eTag = rawHeaders.getValue("ETag");
+        this.xMsRequestId = rawHeaders.getValue("x-ms-request-id");
+        this.xMsClientRequestId = rawHeaders.getValue("x-ms-client-request-id");
+        this.xMsBlobAppendOffset = rawHeaders.getValue("x-ms-blob-append-offset");
+        this.xMsEncryptionScope = rawHeaders.getValue("x-ms-encryption-scope");
+    }
```

Comment:
Have you evaluated whether there is any benefit in lazily computing the header values, moving this logic into the individual getters?

Reply:
Yes, I did do a comparison of eagerly vs lazily evaluating the headers and lazy evaluation performance was better (I'll need to find the result I had) but it was decided to eagerly do all evaluation in the constructor as this would have been a runtime breaking change. Lazy evaluation would push runtime exceptions due to invalid header values to access time instead of the current construction time, if there were a code path that wouldn't be called often I'd push for making the runtime behavior change but since headers are likely to be accessed frequently I'd rather keep current behaviors even if the chances of a corrupted header are low. And another concern is that the headers would be more likely to be mutated with lazy evaluation as they would've left the HttpPipeline and RestProxy scope back into application code.